packages/conform-react/helpers.ts

```diff
@@ -258,13 +257,16 @@ export function getInputProps<Schema, Options extends InputOptions>(
 
 	if (typeof options.value === 'undefined' || options.value) {
 		if (options.type === 'checkbox' || options.type === 'radio') {
-			props.value = typeof options.value === 'string' ? options.value : 'on';
+			props.defaultValue =
+				typeof options.value === 'string' ? options.value : 'on';
 			props.defaultChecked =
 				typeof metadata.initialValue === 'boolean'
 					? metadata.initialValue
-					: metadata.initialValue === props.value;
+					: metadata.initialValue === props.defaultValue;
 		} else if (typeof metadata.initialValue === 'string') {
 			props.defaultValue = metadata.initialValue;
+		} else if (typeof metadata.initialValue === 'number') {
+			props.defaultValue = metadata.initialValue.toString();
```

Comment:
@edmundhung one thing that I was questioning, is should this even be `initialValue`? it seems `value` already does the casting for us via `serialize`

Reply:
I think `serialize` should cast everything to `string` already. So this is probably not needed.

Reply:
if we use `initialValue` its not a string, its the the fully represented type I have patched this functionality locally because previously it was returning nothing for values that were numbers.

Reply:
and also from the #613 PR, we should probably also serialize `bigint` here too

Reply:
The `initialValue` you got from the metadata is already serialized. So once #613 is merged, this should be good.